admin/app/controllers/authenticated/users/list.js

```diff
@@ -16,15 +17,28 @@ export default class ListController extends Controller {
   @tracked firstName = null;
   @tracked lastName = null;
   @tracked email = null;
+
+  @tracked firstNameForm = null;
+  @tracked lastNameForm = null;
+  @tracked emailForm = null;
+
   pendingFilters = {};
 
-  @(task(function* (fieldName, event) {
-    const value = event.target.value;
-    this.pendingFilters[fieldName] = value;
-    yield timeout(this.DEBOUNCE_MS);
-    this.setProperties(this.pendingFilters);
-    this.pendingFilters = {};
+  debouncedUpdateFilters = debounce(this.updateFilters, this.DEBOUNCE_MS);
+
+  updateFilters(filters) {
+    Object.keys(filters).forEach((filterKey) => (this[filterKey] = filters[filterKey]));
     this.pageNumber = DEFAULT_PAGE_NUMBER;
-  }).restartable())
-  triggerFiltering;
+  }
+
+  @action
+  async refreshModel(event) {
+    await this.debouncedUpdateFilters({
+      firstName: this.firstNameForm,
+      lastName: this.lastNameForm,
+      email: this.emailForm,
+    });
```

Comment:
**question**
On ne pourrait pas faire `firstName: this.firstName,` et supprimer les variables avec `Form` ?

Reply:
Il y a des params qui sont bind avec les query params au niveau de la route, d'ou les propriètes suffixées par Form pour dissocier les deux. Et de ne les bind qu'au moment du search.